Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Some tests doesn't produces reproducible results.

It might be some general problem (need to be investigated). Similar but
not the same issue:
http://www.programmersparadox.com/2011/04/20/randomly-failing-php-tests/

--- php-8.0.9/Zend/tests/arginfo_zpp_mismatch.phpt
+++ php-8.0.9/Zend/tests/arginfo_zpp_mismatch.phpt
@@ -3,6 +3,8 @@ Test that there is no arginfo/zpp mismat
 --SKIPIF--
 <?php
 if (getenv('SKIP_MSAN')) die("skip msan misses interceptors for some functions");
+/* on SunOS it sends message via syslog to every logged user (which disturbs them :-) */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
 ?>
 --FILE--
 <?php
--- php-8.0.9/ext/date/tests/bug33532.phpt
+++ php-8.0.9/ext/date/tests/bug33532.phpt
@@ -5,7 +5,7 @@ error_reporting=2047
 date.timezone=UTC
 --SKIPIF--
 <?php
-if(PHP_OS == 'Darwin' || defined('PHP_WINDOWS_VERSION_MAJOR')) die("skip strftime uses system TZ on Darwin and Windows");
+if(PHP_OS == 'Darwin' || PHP_OS == 'SunOS' || defined('PHP_WINDOWS_VERSION_MAJOR')) die("skip strftime uses system TZ on Darwin, Solaris and Windows");
 if (!strftime('%Z')) die('skip strftime does not support %Z');
 ?>
 --FILE--
--- php-8.0.9/ext/openssl/tests/bug77390.phpt
+++ php-8.0.9/ext/openssl/tests/bug77390.phpt
@@ -4,6 +4,8 @@ --SKIPIF--
 <?php
 if (!extension_loaded("openssl")) die("skip openssl not loaded");
 if (!function_exists("proc_open")) die("skip no proc_open");
+/* on SunOS it hangs and never ends */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
 ?>
 --FILE--
 <?php
--- php-8.0.9/ext/standard/tests/file/lstat_stat_variation10.phpt
+++ php-8.0.9/ext/standard/tests/file/lstat_stat_variation10.phpt
@@ -3,6 +3,8 @@ Test lstat() and stat() functions: usage
 --SKIPIF--
 <?php
 if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+/* on SunOS sometime fails */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
 ?>
 --FILE--
 <?php
--- php-8.0.9/ext/standard/tests/file/lstat_stat_variation18.phpt
+++ php-8.0.9/ext/standard/tests/file/lstat_stat_variation18.phpt
@@ -1,5 +1,10 @@
 --TEST--
 Test lstat() and stat() functions: usage variations - dir/file name stored in object
+--SKIPIF--
+<?php
+/* on SunOS sometimes fails - don't know where and why */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
+?>
 --FILE--
 <?php
 /* test for stats of dir/file when their names are stored in objects */
--- php-8.0.9/ext/standard/tests/file/touch_basic.phpt
+++ php-8.0.9/ext/standard/tests/file/touch_basic.phpt
@@ -5,8 +5,8 @@ Dave Kelsey <d_kelsey@uk.ibm.com>
 --SKIPIF--
 <?php
 if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
-if (substr(PHP_OS, 0, 3) == 'WIN') {
-    die('skip.. only for Non Windows');
+if (substr(PHP_OS, 0, 3) == 'WIN' || substr(PHP_OS, 0, 5) == 'SunOS') {
+    die('skip.. only for Non Windows and non SunOS (where it fails sometimes)');
 }
 ?>
 --FILE--
--- php-8.0.9/ext/standard/tests/streams/proc_open_bug60120.phpt
+++ php-8.0.9/ext/standard/tests/streams/proc_open_bug60120.phpt
@@ -1,5 +1,12 @@
 --TEST--
 Bug #60120 proc_open hangs with stdin/out with >2048 bytes
+--SKIPIF--
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS sometimes fails - don't know where and why */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 error_reporting(E_ALL);
--- php-8.0.9/ext/standard/tests/streams/proc_open_bug64438.phpt
+++ php-8.0.9/ext/standard/tests/streams/proc_open_bug64438.phpt
@@ -1,8 +1,14 @@
 --TEST--
 Bug #64438 proc_open hangs with stdin/out with 4097+ bytes
+--SKIPIF--
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS sometimes fails - don't know where and why */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
-
 error_reporting(E_ALL);
 
 if (substr(PHP_OS, 0, 3) == 'WIN') {
--- php-8.0.9/sapi/fpm/tests/bug68381-log-level-warning.phpt
+++ php-8.0.9/sapi/fpm/tests/bug68381-log-level-warning.phpt
@@ -4,6 +4,8 @@ --SKIPIF--
 <?php
 include "skipif.inc";
 if (getenv("SKIP_ASAN")) die("skip Often fails on azure with asan for unknown reason");
+/* on SunOS it newer ends. */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/bug68420-ipv4-all-addresses.phpt
+++ php-8.0.9/sapi/fpm/tests/bug68420-ipv4-all-addresses.phpt
@@ -3,6 +3,10 @@ FPM: bug68420 - IPv4 all addresses
 --SKIPIF--
 <?php
 include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/bug68442-signal-reload.phpt
+++ php-8.0.9/sapi/fpm/tests/bug68442-signal-reload.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug68442 - Signal reload
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug68458-pm-no-start-server.phpt
+++ php-8.0.9/sapi/fpm/tests/bug68458-pm-no-start-server.phpt
@@ -3,6 +3,10 @@ FPM: bug68458 - Missing pm.start_servers
 --SKIPIF--
 <?php
 include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/bug69625-no-script-filename.phpt
+++ php-8.0.9/sapi/fpm/tests/bug69625-no-script-filename.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug69625 - 404 should be returned on missing SCRIPT_FILENAME
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug72573-http-proxy.phpt
+++ php-8.0.9/sapi/fpm/tests/bug72573-http-proxy.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug72573 - HTTP_PROXY - CVE-2016-5385
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it hangs and never ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug73342-nonblocking-stdio.phpt
+++ php-8.0.9/sapi/fpm/tests/bug73342-nonblocking-stdio.phpt
@@ -1,7 +1,10 @@
 --TEST--
 FPM: bug73342 - Non-blocking stdin
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php include "skipif.inc";
+/* on SunOS it hangs and never ends */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug74083-concurrent-reload.phpt
+++ php-8.0.9/sapi/fpm/tests/bug74083-concurrent-reload.phpt
@@ -2,8 +2,10 @@ --TEST--
 Concurrent reload signals should not kill PHP-FPM master process. (Bug: #74083)
 --SKIPIF--
 <?php
-include "skipif.inc";
-if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/bug75212-php-value-in-user-ini.phpt
+++ php-8.0.9/sapi/fpm/tests/bug75212-php-value-in-user-ini.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug75212 - php_value acts like php_admin_value
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug76601-reload-child-signals.phpt
+++ php-8.0.9/sapi/fpm/tests/bug76601-reload-child-signals.phpt
@@ -2,9 +2,10 @@ --TEST--
 FPM: bug76601 children should not ignore signals during concurrent reloads
 --SKIPIF--
 <?php
-include "skipif.inc";
-if (getenv("SKIP_SLOW_TESTS")) die("skip slow test");
-?>
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug77934-reload-process-control.phpt
+++ php-8.0.9/sapi/fpm/tests/bug77934-reload-process-control.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug77934 - php-fpm kill -USR2 not working
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/bug80024-socket-reduced-inherit.phpt
+++ php-8.0.9/sapi/fpm/tests/bug80024-socket-reduced-inherit.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: bug80024 - Duplication of info about inherited socket after pool removing
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/log-bwd-msg-with-nl.phpt
+++ php-8.0.9/sapi/fpm/tests/log-bwd-msg-with-nl.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: Buffered worker output decorated log with msg containing new lines
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it hangs and never ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/log-bwd-multiple-msgs-stdout-stderr.phpt
+++ php-8.0.9/sapi/fpm/tests/log-bwd-multiple-msgs-stdout-stderr.phpt
@@ -1,7 +1,10 @@
 --TEST--
 FPM: Buffered worker output decorated log with multiple continuous messages (stdout/stderr mixed)
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+/* on SunOS it hangs and never ends */
+if (PHP_OS == 'SunOS') die('skip.. Not valid for SunOS');
+?>
 --XFAIL--
 Fails intermittently
 --FILE--
--- php-8.0.9/sapi/fpm/tests/log-bwd-multiple-msgs.phpt
+++ php-8.0.9/sapi/fpm/tests/log-bwd-multiple-msgs.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: Buffered worker output decorated log with multiple continuous messages
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it hangs and never ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/reload-uses-sigkill-as-last-measure.phpt
+++ php-8.0.9/sapi/fpm/tests/reload-uses-sigkill-as-last-measure.phpt
@@ -2,9 +2,10 @@ --TEST--
 If SIGQUIT and SIGTERM during reloading fail, SIGKILL should be sent
 --SKIPIF--
 <?php
-include "skipif.inc";
-if (!function_exists('pcntl_sigprocmask')) die('skip Requires pcntl_sigprocmask()');
-if (getenv('SKIP_SLOW_TESTS')) die('skip slow tests excluded by request');
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it kills all tests by the signal */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/socket-invalid-allowed-clients.phpt
+++ php-8.0.9/sapi/fpm/tests/socket-invalid-allowed-clients.phpt
@@ -2,7 +2,10 @@ --TEST--
 FPM: Socket for invalid allowed client only
 --SKIPIF--
 <?php
-include "skipif.inc";
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it hangs and never ends */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
--- php-8.0.9/sapi/fpm/tests/socket-uds-basic.phpt
+++ php-8.0.9/sapi/fpm/tests/socket-uds-basic.phpt
@@ -1,7 +1,12 @@
 --TEST--
 FPM: Unix Domain Socket connection
 --SKIPIF--
-<?php include "skipif.inc"; ?>
+<?php
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
+?>
 --FILE--
 <?php
 
--- php-8.0.9/sapi/fpm/tests/socket-uds-numeric-ugid-nonroot.phpt
+++ php-8.0.9/sapi/fpm/tests/socket-uds-numeric-ugid-nonroot.phpt
@@ -5,6 +5,10 @@ <?php
 include "skipif.inc";
 FPM\Tester::skipIfRoot();
 FPM\Tester::skipIfPosixNotLoaded();
+if (substr(PHP_OS, 0, 5) == 'SunOS') {
+    /* on SunOS it newer ends */
+    die('skip.. Not valid for SunOS');
+}
 ?>
 --FILE--
 <?php
